### PR TITLE
GRIDEDIT-800: bind mesh transformation functions

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 1]
```diff
@@ -1172,9 +1172,27 @@ public interface IMeshKernelApi : IDisposable
...
         int Mesh2dRefineBasedOnSamples(int meshKernelId, in DisposableGeometryList disposableGeometryListIn,
                                        double relativeSearchRadius, int minimumNumSamples, in MeshRefinementParameters meshRefinementParameters);
+        /// <summary>
+        /// Rotates a mesh2d about a given point by a given angle
+        /// </summary>
+        /// <param name="meshKernelId">Id of the grid state</param>
+        /// <param name="centreX">X-coordinate of the centre of rotation</param>
+        /// <param name="centreY">Y-coordinate of the centre of rotation</param>
+        /// <param name="angle">Angle of rotation</param>
+        /// <returns>Error code</returns>
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:1185]: angle in radians I suppose? suggest to add it to the documentation.

File: src/MeshKernelNET/Native/MeshKernelDll.cs [hunk 2]
```diff
@@ -1433,6 +1433,32 @@ static MeshKernelDll()
                                                               [In] int minimumNumSamples,
                                                               [In] ref MeshRefinementParametersNative meshRefinementParameters);
 
+        /// <summary>
+        /// Rotates a mesh2d about a given point by a given angle
+        /// </summary>
+        /// <param name="meshKernelId">Id of the mesh state</param>
+        /// <param name="centreX">>X-coordinate of the centre of rotation</param>
+        /// <param name="centreY">Y-coordinate of the centre of rotation></param>
+        /// <param name="angle">Angle of rotation</param>
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Native/MeshKernelDll.cs:1442]: suggest to document that angle is in radians or degrees

**Review by andreasbuykx — APPROVED** [review]
Looks good. I think documenting the unit of angle is a good idea, I made the incorrect assumption during the review ...

File: src/MeshKernelNET/Api/IMeshKernelApi.cs (hunk 1 above)
Comment (ahmad-el-sayed) [reply src/MeshKernelNET/Api/IMeshKernelApi.cs:1185]: Degrees actually. I'll add units everywhere.

File: src/MeshKernelNET/Native/MeshKernelDll.cs (hunk 2 above)
Comment (ahmad-el-sayed) [reply src/MeshKernelNET/Native/MeshKernelDll.cs:1442]: Degrees. will add it.

**Review by andreasbuykx — APPROVED** [review]

**Pull request merged**